mm2src/mm2_main/tests/mm2_tests/mm2_tests_inner.rs

```diff
@@ -7927,3 +7950,342 @@ fn test_eth_swap_negotiation_fails_maker_no_fallback() {
     block_on(wait_for_swap_negotiation_failure(&mm_bob, &uuids[0], wait_until));
     block_on(wait_for_swap_negotiation_failure(&mm_alice, &uuids[0], wait_until));
 }
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields, tag = "status", content = "details")]
+pub enum InitTrezorStatus {
+    Ok(InitHwResponse),
+    Error(Json),
+    InProgress(Json),
+    UserActionRequired(Json),
+}
+
+pub fn mm_ctx_with_trezor(conf: Json) -> MmArc {
+    let ctx = mm_ctx_with_custom_db_with_conf(Some(conf));
+
+    CryptoCtx::init_with_iguana_passphrase(ctx.clone(), "123456").unwrap(); // for now we need passphrase seed for init
+    let req: InitHwRequest = serde_json::from_value(json!({ "device_pubkey": null })).unwrap();
+    let res = match block_on(init_trezor(ctx.clone(), req)) {
+        Ok(res) => res,
+        _ => {
+            panic!("cannot init trezor");
+        },
+    };
+
+    block_on(async {
+        loop {
+            let status_req = RpcTaskStatusRequest {
+                task_id: res.task_id,
+                forget_if_finished: false,
+            };
+            match init_trezor_status(ctx.clone(), status_req).await {
+                Ok(res) => {
+                    log!("trezor init status={:?}", serde_json::to_string(&res).unwrap());
+                    match res {
+                        RpcTaskStatus::Ok(_) => {
+                            log!("device initialized");
+                            break;
+                        },
+                        RpcTaskStatus::Error(_) => {
+                            log!("device in error state");
+                            break;
+                        },
+                        RpcTaskStatus::InProgress(_) => log!("trezor init in progress"),
+                        RpcTaskStatus::UserActionRequired(_) => log!("device is waiting for user action"),
+                    }
+                },
+                _ => {
+                    panic!("cannot get trezor status");
+                },
+            };
+            Timer::sleep(5.).await
+        }
+    });
+    ctx
+}
+
+/// Tool to run withdraw directly with trezor device or emulator (no rpc version, added for easier debugging)
+/// run cargo test with '--ignored' option
+/// to use trezor emulator add '--features trezor-udp' option to cargo test params
+#[test]
+#[ignore]
+#[cfg(all(not(target_arch = "wasm32")))]
+fn test_withdraw_from_trezor_segwit_no_rpc() {
+    use common::wait_until_ms;
+
+    let ticker = "tBTC-segwit";
+    let coin_conf = json!({
```

Comment:
You can use `tBTC-segwit` conf function here too.

Reply:
done